### PR TITLE
Adding maximumFaultDomainCountSupported and currentCapacity property for the Capacity Reservation Resource in API Version 2022-08-01 

### PR DESCRIPTION
**Pull request by grizzlytheodore**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [x] SDK of Java
      - [x] SDK of Js
      - [x] SDK of Go
      - [x] PowerShell
      - [x] CLI
      - [x] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2022-08-01/capacityReservation.json [hunk 1]
```diff
@@ -873,6 +873,12 @@
           "type": "string",
           "description": "A unique id generated and assigned to the capacity reservation by the platform which does not change throughout the lifetime of the resource."
         },
+        "platformFaultDomainCount": {
```
Comment (grizzlytheodore) [inline specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2022-08-01/capacityReservation.json:876]: @ArcturusZhang  there is a ["platformFaultDomainCount" property set in virtualmachinescaleset.json ](https://github.com/Azure/azure-rest-api-specs/blob/feature/cplat-2022-08-01/specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2022-08-01/virtualMachineScaleSet.json#L4739) but it is not readonly. 
can an object with same name defined here but with readonly tag?

**Comment by grizzlytheodore** [comment]
for some reason, it's not letting me set this as approved, but approving verbally

File: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2022-08-01/capacityReservation.json [hunk 2]
```diff
@@ -838,6 +838,12 @@
     "CapacityReservationUtilization": {
       "type": "object",
       "properties": {
+        "currentCapacity": {
+          "readOnly": true,
+          "type": "integer",
+          "format": "int32",
+          "description": "The value provides the actual reserved count for the VM size which was provisioned sucessfully and for which the customer is getting billed for the Capacity Reservation.<br><br>Minimum api-version: 2022-08-01."
```
Comment (smotwani) [inline specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2022-08-01/capacityReservation.json:845]: The value provides the current capacity of the VM size which was reserved successfully and for which the customer is getting billed.<br><br>Minimum api-version: 2022-08-01.

File: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2022-08-01/examples/capacityReservationExamples/CapacityReservation_Update_MaximumSet_Gen.json [hunk 3]
```diff
@@ -19,7 +19,7 @@
         }
       },
       "sku": {
-        "name": "DSv3-Type1",
+        "name": "Standard_DS1_v2",
         "tier": "aaa",
         "capacity": 7
```
Comment (smotwani) [inline specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2022-08-01/examples/capacityReservationExamples/CapacityReservation_Update_MaximumSet_Gen.json:24]: we don't use the tier property in request or response, so we should remove it from the example.

File: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2022-08-01/capacityReservation.json [hunk 4]
```diff
@@ -873,6 +879,12 @@
           "type": "string",
           "description": "A unique id generated and assigned to the capacity reservation by the platform which does not change throughout the lifetime of the resource."
         },
+        "platformFaultDomainCount": {
+          "readOnly": true,
+          "type": "integer",
+          "format": "int32",
+          "description": "This is the readonly property that is populated and set in the response to specify the Maximum Fault Domain Count that capacity reservation support for the VM Size.<br><br>Minimum api-version: 2022-08-01."
```
Comment (smotwani) [inline specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2022-08-01/capacityReservation.json:886]: No need to mention this is a read only property it is already specified via the readOnly attribute set on property in the spec, let us update the comment to "Specifies the value of fault domain count that Capacity Reservation supports for requested VM Size. <br>NOTE:  The fault domain count of a resource that supports Capacity Reservation must less than or equal to this value.

File: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2022-08-01/examples/capacityReservationExamples/CapacityReservationGroup_Get.json [hunk 5]
```diff
@@ -23,6 +23,7 @@
               {
                 "name": "myCapacityReservation1",
                 "utilizationInfo": {
+                  "currentCapacity": 5,
```
Comment (smotwani) [inline specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2022-08-01/examples/capacityReservationExamples/CapacityReservationGroup_Get.json:26]: In this and other example files for capacity reservation change expand parameter wherever specified to $expand.

File: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2022-08-01/examples/capacityReservationExamples/CapacityReservation_Get.json [hunk 6]
```diff
@@ -10,6 +10,7 @@
     "200": {
       "body": {
         "properties": {
+          "platformFaultDomainCount": 3,
```
Comment (smotwani) [inline specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2022-08-01/examples/capacityReservationExamples/CapacityReservation_Get.json:13]: pls. add the $expand = instanceView parameter in the request parameters as this example is also returning the instance view in the response.

File: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2022-08-01/examples/capacityReservationExamples/CapacityReservation_Update_MaximumSet_Gen.json (hunk 3 above)
Comment (smotwani) [reply specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2022-08-01/examples/capacityReservationExamples/CapacityReservation_Update_MaximumSet_Gen.json:24]: "tier": "aaa" is still there in the example, pls. remove as we don't use it.

**Review by smotwani — APPROVED** [review]

**Pull request merged**